hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/checksum/ContainerChecksumTreeManager.java

```diff
@@ -44,12 +46,15 @@ public class ContainerChecksumTreeManager {
   // Used to coordinate reads and writes to each container's checksum file.
   // Each container ID is mapped to a stripe.
   private final Striped<ReadWriteLock> fileLock;
+  private final ContainerMerkleTreeMetrics metrics;
 
   /**
    * Creates one instance that should be used to coordinate all container checksum info within a datanode.
    */
   public ContainerChecksumTreeManager(DatanodeConfiguration dnConf) {
     fileLock = SimpleStriped.readWriteLock(dnConf.getContainerChecksumLockStripes(), true);
+    // TODO: TO unregister metrics on stop.
```

Comment:
I made a note of this [here](https://github.com/apache/ozone/pull/6875/files#r1688351623). This PR connects the manager to `OzoneContainer` where we can unregister the metrics.

Reply:
Yes we need to add it there